flink-doris-connector/src/main/java/org/apache/doris/flink/table/DorisDynamicTableSource.java

```diff
@@ -212,5 +200,17 @@ public boolean supportsNestedProjection() {
     @Override
     public void applyProjection(int[][] projectedFields, DataType producedDataType) {
         this.physicalRowDataType = Projection.of(projectedFields).project(physicalRowDataType);
+        if (StringUtils.isNullOrWhitespaceOnly(readOptions.getFilterQuery())) {
+            String filterQuery = resolvedFilterQuery.stream().collect(Collectors.joining(" AND "));
+            this.readOptions.setFilterQuery(filterQuery);
+        }
+        if (StringUtils.isNullOrWhitespaceOnly(readOptions.getReadFields())) {
+            String[] selectFields =
+                    DataType.getFieldNames(physicalRowDataType).toArray(new String[0]);
+            this.readOptions.setReadFields(
+                    Arrays.stream(selectFields)
+                            .map(item -> String.format("`%s`", item.trim().replace("`", "")))
+                            .collect(Collectors.joining(", ")));
+        }
```

Comment:
What are the benefits of putting it below?

Reply:
The method `getScanRuntimeProvider` is executed before `applyProjection`. Therefore, initializing `readFields` in `getScanRuntimeProvider` will result in querying all fields in Doris, leading to type conversion errors.